src/js/node/Report.hx

```diff
@@ -0,0 +1,54 @@
+package js.node;
+
+#if haxe4
+import js.lib.Error;
+#else
+import js.Error;
+#end
+
+extern class Report {
+	/**
+		If the reports are written in compact mode.
+	**/
+	var compact:Bool;
+	
+	/**
+		The directory in which the reports are written.
+	**/
+	var directory:String;
+	
+	/**
+		The file in which the reports are written.
+	**/
+	var filename:String;
+	
+	/**
+		Returns the diagnostic report as an object. Optionally from an error object.
+	**/
+	function getReport(?err:Error):Dynamic;
+	
+	/**
+		When the diagnostic report was generated in case of fatal errors.
+	**/
+	var reportOnFatalError:Bool;
+	
+	/**
+		If the diagnostic report was generated by receiving a signal.
+	**/
+	var reportOnSignal:Bool;
+	
+	/**
+		When the diagnostic report was generated in case of a uncaught exception.
+	**/
+	var reportOnUncaughtException:Bool;
+	
+	/**
+		The signal that triggered the diagnostic report.
+	**/
+	var signal:String;
+	
+	/**
+		Writes a diagnostic report to a file.
+	**/
+	function writeReport(?filename:String, ?err:Error):Void;

```

Comment:
Return type should be `String` https://nodejs.org/docs/latest-v12.x/api/process.html#process_process_report_writereport_filename_err